cylc/uiserver/handlers.py

```diff
@@ -28,7 +30,55 @@
 from .websockets import authenticated as websockets_authenticated
 
 
-class BaseHandler(web.RequestHandler):
+logger = logging.getLogger(__name__)
+
+
+ME = getpass.getuser()
```

Comment:
Not sure if this would work for every JupyterHub authenticator? Maybe for some users the user database is stored in a database, external system, etc. So maybe `getpass.getuser()` would lock us to linux & pam only?

Reply:
In order for the UIS to run the user has to exist on the system so it should be good for most things but in more esoteric systems where the authenticated user does not match any system user it would break yes (although I'm not sure how the hub could spawn UI Servers in this case).

We will add the ability to configure authorisation (i.e. grant access to different accounts) that should help us to handle this sort of thing.

Reply:
Maybe we could use Tornado's [`current_user`](https://www.tornadoweb.org/en/stable/guide/security.html#user-authentication) then?

When the hub is configured to use PAM, it will match what user you have logged in. Then we can compare that against the request path.

So if I logged in with a database backed authenticator and have the user `kinow`, but I'm accessing the URL `/#/workflows/userx|one`, then our code to authorise would just match `current_user` (`kinow` from the database in this case, but would work with PAM, etc) and fail if I don't have authorization I think?

Also, if we run Cylc UIS with `root` or another superuser, what would `getpass.getuser()` return? Wouldn't it return `root` instead of the logged-in user?

Reply:
> Also, if we run Cylc UIS with root or another superuser

The UIS should never be run as root. It is only run under a regular system user.

Reply:
I think `current_user` is the result of `get_current_user` which returns the authenticated user who performed the request.

We need to know the user that the UIS is running under.

Reply:
>The UIS should never be run as root. It is only run under a regular system user.

I thought BoM was considering using a system user with elevated privileges for the UIS.

Reply:
I think that's for the Hub?

It should still work for an account with elevated privileges, but they would need to authenticate with the Hub as the elevated user, the hub would then spawn the UIS as that user.

Reply:
Ah yeah, you are correct, the UIS should be with my own user, my bad!

Reply:
>  in more esoteric systems where the authenticated user does not match any system user it would break yes (although I'm not sure how the hub could spawn UI Servers in this case).

Not sure that's true.  The authenticator doesn't have to know about users on the target system where the UIS and Schedulers run.  So long as our authorization system doesn't require the authenticated user to be valid in its domain, we should be able to authorize "alien" users to access the workflows, no?

Reply:
(We do of course have to trust that the authentication system has been configured safely).

Reply:
> we should be able to authorize "alien" users to access the workflows, no?

Yes, however, those users won't be able to start UI Servers.

(This thread was about the use of `getpass.getuser()` in the UIS for determining the name of UIS owner.)

Reply:
OK, you're talking about the current PR, not where we're heading (any authenticated user can spawn a UIS on other users' accounts).